designs/2019-mjs-cjs-sourceType-override/README.md

```diff
@@ -0,0 +1,47 @@
+- Start Date: 2019-12-16
+- RFC PR: https://github.com/eslint/rfcs/pull/53/
+- Authors: LJ Talbot ([@LJNeon](https://github.com/LJNeon/))
+
+# Files with .mjs or .cjs extensions should override sourceType
+
+## Summary
+
+Files with a `.mjs` extension are always modules and files with a `.cjs` extension are always scripts. The `sourceType` configuration option currently causes eslint to handle those files incorrectly.
+
+## Motivation
+
+Node.js support for esm is now unflagged in version 13, and the ability to support both cjs and esm modules was also added. As node.js esm support stabilities the amount of packages that use this will grow significantly. This will introduce code that does not utilize only one `sourceType`, and node.js handles this via `.mjs` and `.cjs` file extensions. This is guaranteed behavior and should be supported.
+
+## Detailed Design
+
+This would be implemented by changing `sourceType` to only affect files with a `.js` file extension. Files with a `.mjs` extension are always treated as modules, and files with a `.cjs` extension are always treated as scripts.
```

Comment:
Would you describe more detailed?

- How this proposal affects the Node.js API of ESLint? What classes/methods should be changed?
- How this proposal affects the authors of plugins or custom parsers? For example, ESLint doesn't decide the content of `parserOptions` and it's up to custom parser authors completely. In other words, `parserOptions.sourceType` doesn't mean the option that changes parsing mode. Does this proposal change it?